service/collector/vpc.go

```diff
@@ -0,0 +1,93 @@
+package collector
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	ClusterTag      = "giantswarm.io/cluster"
+	InstallationTag = "giantswarm.io/installation"
+	NameTag         = "Name"
+	OrganizationTag = "giantswarm.io/organization"
+	StackNameTag    = "aws:cloudformation:stack-name"
+
+	CidrLabel         = "cidr"
+	ClusterLabel      = "cluster_id"
+	IDLabel           = "id"
+	InstallationLabel = "installation"
+	NameLabel         = "name"
+	OrganizationLabel = "organization"
+	StackNameLabel    = "stack_name"
+	StateLabel        = "state"
+)
+
+var (
+	vpcs *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "", "vpc_info"),
+		"VPC information.",
+		[]string{
+			CidrLabel,
+			ClusterLabel,
+			IDLabel,
+			InstallationLabel,
+			NameLabel,
+			OrganizationLabel,
+			StackNameLabel,
+			StateLabel,
+		},
+		nil,
+	)
+)
+
+func (c *Collector) collectVPCs(ch chan<- prometheus.Metric) {
+	c.logger.Log("level", "debug", "message", "collecting metrics for vpcs")
+
+	resp, err := c.awsClients.EC2.DescribeVpcs(&ec2.DescribeVpcsInput{})
+	if err != nil {
+		c.logger.Log("level", "error", "message", "could not list vpcs", "error", fmt.Sprintf("%#v", err))
```

Comment:
The `stack` field is already used in other projects. Making up your own thing now will not help people debugging via EFK. I do not have strong feelings about the identifier here. The important thing for me is that we use the same identifier for one thing everywhere to make our lives easier. Feel free to change this everywhere though, but changing this in one project and keeping the others unchanged is not going to work that well for humans.

Reply:
For whoever comes later, see https://github.com/giantswarm/operatorkit/blob/c1a1a0c6c139450e5db7c92e92fb8706984a0cf9/framework/resource/retryresource/crud_resource_ops_wrapper.go#L71 for example